providers/snowflake/warehouse_grant.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 The Terraformer Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package snowflake
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/terraformer/terraform_utils"
+)
+
+type WarehouseGrantGenerator struct {
+	SnowflakeService
+}
+
+func (g WarehouseGrantGenerator) createResources(warehouseGrantList []warehouseGrant) []terraform_utils.Resource {
+	type tfGrant struct {
```

Comment:
we're using this almost everywhere, it probably means we should make it an actual top level type